Add support for Symfony 3

mbabker opened the pull request:
This PR adds support for Symfony 3.0.  In doing so, this also addresses the B/C breaking change regarding the form type for Symfony 3.0 where the FQCN is expected.

egeloen commented:
Nice! It was on my TODO list but I have not find time to do it. Anyway, my plan was also to include the `symfony/phpunit-bridge` in order to more easily detect such deprecations. Can you introduce it as well?

mbabker commented:
Added it and fixed a small bug.  Looks like there's an issue with Packagist and the `symfony/form` package as right now it isn't listing any of the stable tags, so builds will fail for the moment.

mbabker commented:
Packagist is working fine.  Aside from a timeout downloading the dependencies on the 2.6 run, all's well.

egeloen commented:
Thanks for your work! I have re-think about these deprecations and there is one which is currently not handle. The service tag attr `alias` of the `form.type` has not been removed for Symfony >= 2.8. IMO, we should introduce a compile pass which add or remove the tag attr conditionally according tot he kernel version.

mbabker commented:
That should be covered now.  Would've been nice to see it in their upgrade documentation somewhere.

egeloen commented:
I have squash and cherry pick your commit in 9e4ae7e675ac0fff9a829e5fe400b4a8c2e2802c Thanks!

mbabker commented:
:+1:
